src/Position.js

```diff
@@ -49,55 +51,62 @@ class Position extends React.Component {
     );
   }
 
-  _maybeUpdatePosition(){
-    if ( this._needsFlush ) {
-      this._needsFlush = false;
-      this._updatePosition();
+  getTargetSafe(props) {
+    if (!props.target) {
+      return null;
     }
+
+    return props.target(props);
```

Comment:
I think it's a little more clear to write "exceptional" cases as traps at the start of a function rather than branches, e.g. I always use something like

```js
if (thingsAreMessedUp) {
  return;
}

// do actual work
```